Add RIRO and Static masks to realtime zshimming

po09i opened the pull request:
## Description
With the goal of improving our realtime zshimming workflow, we can enhance our masking capability. We currently mask the shimming region we want to improve. Adding the ability to mask the riro component and the static component independently would be a nice feature and a lead to further investigations.

This PR adds `nii_mask_anat_static` and `nii_mask_anat_riro` to the `realtime_zshim()` API and appropriately masks the relevant components. The PR also adds `mask_riro` and `mask_static` options to the `st_realtime_zshim` CLI. The relevant tests were modified and the demo shell script was updated.

## Linked issues
Fixes #220

po09i committed:
Update to input static and riro mask

po09i committed:
Update tests for rtzshim API

po09i committed:
Update the CLI and tests for the CLI

po09i committed:
Update the shell script to input both masks

evaalonsoortiz commented:
I am trying to test this out an getting the following error:

```
The filename for the output mask is: /Users/evaalonsoortiz/Documents/Academic/Postdoc_Julien/Projects/Realtime_zShimming/Data/acdc_123p/rt_zshim_nifti/sub-example/fmap/sub-example_anat_mask.nii.gz
Usage: st_realtime_zshim [OPTIONS]
Try 'st_realtime_zshim -h' for help.

Error: no such option: -m
```

Note: I'm not using -m as an option in the call to `st_realtime_zshim`

po09i commented:
Can you post the shell call you are using so I can try to reproduce the error

evaalonsoortiz commented:
> Can you post the shell call you are using so I can try to reproduce the error

`st_prepare_fieldmap "sub-example_phasediff.nii.gz" -mag "sub-example_magnitude1.nii.gz" -unwrapper "prelude" -output "sub-example_fieldmap.nii.gz" -gaussian_filter=True || exit`

`st_mask box -input "../anat/sub-example_unshimmed_e1.nii.gz" -size 20 20 30 -center 60 32 7 -output "sub-example_anat_mask.nii.gz" || exit
`

`st_realtime_zshim -fmap "sub-example_fieldmap.nii.gz" -anat "../anat/sub-example_unshimmed_e1.nii.gz" -resp "../../../realtime_zshimming_data/PMUresp_signal.resp" -mask_anat "sub-example_anat_mask.nii.gz" -mask_riro "sub-example_anat_mask.nii.gz" -output "." || exit
`

jcohenadad commented:
@evaalonsoortiz no need for the "" in the syntax (quicker to type without them 😊 )

po09i committed:
Fix typo for shell script

po09i commented:
@evaalonsoortiz There was a typo in the shell script, the option was -mask_anat but it should have been -mask_static. I corrected it and it should work now.

jcohenadad commented:
i didn't realize there was a mix of "-" and "_" in CLI flags. We should stick to "-" only-- can someone pls open an issue @charjl @po09i

jcohenadad commented on shimmingtoolbox/cli/realtime_zshim.py:
```diff
@@ -22,13 +22,16 @@
+              help="3D nifti file used to define the static spatial region to shim. "
+                   "The coordinate system should be the same as ``anat``'s coordinate system.")
+@click.option('-mask_riro', 'fname_mask_anat_riro', type=click.Path(), required=False,
+              help="3D nifti file used to define the time varying spatial region to shim. "
```
spell out RIRO in the help

jcohenadad commented on shimmingtoolbox/cli/realtime_zshim.py:
```diff
@@ -22,13 +22,16 @@
               help="Siemens respiratory file containing pressure data.")
-@click.option('-mask', 'fname_mask_anat', type=click.Path(), required=False,
-              help="3D nifti file with used to define the spatial region to shim. "
+@click.option('-mask_static', 'fname_mask_anat_static', type=click.Path(), required=False,
```
```suggestion
@click.option('-mask-static', 'fname_mask_anat_static', type=click.Path(), required=False,
```

jcohenadad commented on shimmingtoolbox/cli/realtime_zshim.py:
```diff
@@ -22,13 +22,16 @@
+@click.option('-mask_static', 'fname_mask_anat_static', type=click.Path(), required=False,
+              help="3D nifti file used to define the static spatial region to shim. "
+                   "The coordinate system should be the same as ``anat``'s coordinate system.")
+@click.option('-mask_riro', 'fname_mask_anat_riro', type=click.Path(), required=False,
```
```suggestion
@click.option('-mask-riro', 'fname_mask_anat_riro', type=click.Path(), required=False,
```

evaalonsoortiz commented:
> @evaalonsoortiz There was a typo in the shell script, the option was -mask_anat but it should have been -mask_static. I corrected it and it should work now.

Just gave it another go, but still getting the same error ...

po09i committed:
Updated option names and added RIRO comment

po09i committed:
Merge master

po09i commented:
> > @evaalonsoortiz There was a typo in the shell script, the option was -mask_anat but it should have been -mask_static. I corrected it and it should work now.
> 
> Just gave it another go, but still getting the same error ...

Problem is now fixed. The error was the one described above [here](https://github.com/shimming-toolbox/shimming-toolbox/pull/221#issuecomment-774254852)

evaalonsoortiz reviewed (approved):
Tested it out, looks good.

po09i committed:
Resolve conflcts with master
